integration_tests/auth/test_auth.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime, timedelta, timezone
+from time import sleep
+from typing import Any, Callable, Dict, Optional
+
+import pytest
+from faker import Faker
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+from opentelemetry.trace import Span
+from phoenix.auth import AUTH_HEADER
+from typing_extensions import TypeAlias
+
+ProjectName: TypeAlias = str
+SpanName: TypeAlias = str
+Headers: TypeAlias = Dict[str, Any]
+Name: TypeAlias = str
+ApiKey: TypeAlias = str
+
+NOW = datetime.now(timezone.utc)
+
+
+class TestSpanExporters:
+    @pytest.mark.parametrize(
+        "with_headers,expires_at,expected",
+        [
+            (True, NOW + timedelta(days=1), SpanExportResult.SUCCESS),
+            (True, None, SpanExportResult.SUCCESS),
+            (True, NOW, SpanExportResult.FAILURE),
+            (False, None, SpanExportResult.FAILURE),
+        ],
+    )
+    def test_headers(
+        self,
```

Comment:
Are the tests running concurrently or one after another?

Reply:
`pytest` doesn't have concurrency, but we use `pytest-xdist`, so you'll see several different port numbers in the log for this test.